plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerAnimator.java

```diff
@@ -26,41 +30,56 @@ void removeListener(OnAnimationsValuesChangeListener listener) {
     listeners.remove(listener);
   }
 
-  void feedNewLocation(@NonNull Location previousLocation, @NonNull Location newLocation) {
-    LatLng previousLatLng;
-    if (latLngAnimator != null) {
-      previousLatLng = (LatLng) latLngAnimator.getAnimatedValue();
-    } else {
-      previousLatLng = new LatLng(previousLocation);
+  void feedNewLocation(@NonNull Location newLocation) {
+    if (previousLocation == null) {
+      previousLocation = newLocation;
     }
-    LatLng newLatLng = new LatLng(newLocation);
 
-    float previousBearing;
-    if (gpsBearingAnimator != null) {
-      previousBearing = (float) gpsBearingAnimator.getAnimatedValue();
-    } else {
-      previousBearing = previousLocation.getBearing();
-    }
+    LatLng previousLatLng = getPreviousLatLng();
+    LatLng targetLatLng = new LatLng(newLocation);
+
+    float previousBearing = getPreviousGpsBearing();
+    float targetBearing = newLocation.getBearing();
+
+    // TODO create animator duration / interpolator
+//    float speed = location == null ? 0 : location.getSpeed();
+//
+//    locationChangeAnimator.setDuration(linearAnimation || speed > 0
+//      ? getLocationUpdateDuration() : LocationLayerConstants.LOCATION_UPDATE_DELAY_MS);
+//    if (linearAnimation || speed > 0) {
+//      locationChangeAnimator.setInterpolator(new LinearInterpolator());
+//    } else {
+//      locationChangeAnimator.setInterpolator(new AccelerateDecelerateInterpolator());
+//    }
 
     cancelLocationAnimations();
-    latLngAnimator = new LatLngAnimator(previousLatLng, newLatLng, 1000);
-    gpsBearingAnimator = new BearingAnimator(previousBearing, newLocation.getBearing(), 1000);
-    // FIXME: 22/02/2018 evaluate duration of animation better
+    latLngAnimator = new LatLngAnimator(previousLatLng, targetLatLng, 1000);
```

Comment:
can we replace the constant 1000 with the [following](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/widgets/MyLocationView.java#L998-L1012) logic for now. This should make the animation a bit more continuous.